dashboard/app/models/hint_view_request.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: hint_view_requests
+#
+#  id            :integer          not null, primary key
+#  user_id       :integer
+#  script_id     :integer
+#  level_id      :integer
+#  feedback_type :integer
+#  feedback_xml  :text(65535)
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+# Indexes
+#
+#  index_hint_view_requests_on_script_id_and_level_id  (script_id,level_id)
+#  index_hint_view_requests_on_user_id                 (user_id)
+#
+
+class HintViewRequest < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :script
+  belongs_to :level
+
+  validates :user, :presence => true
+  validates :script, :presence => true
+  validates :level, :presence => true
+
+  ENABLED = true
+
+  # Returns an array of serializable hashes representing all of a given
+  # user's hint view requests for a given script and level
+  # called by ApplicationController.milestone_response
+  # used by Studio's feedback modal to decide whether or not to show a
+  # user a hint based on the already-viewed hints.
+  def HintViewRequest.milestone_response(script, level, user)
+    HintViewRequest::ENABLED ? HintViewRequest.where(script: script, level: level, user: user)
+      .pluck(:feedback_type, :feedback_xml)
+      .map do |feedback_type, feedback_xml|
+        {feedback_type: feedback_type, feedback_xml: feedback_xml}
+      end : []
+  end
+end
```

Comment:
weird, try moving the . for multiline chains (like in `.pluck`) to the previous line.

when loading executing files they are the same (but this changed, in ruby 1.9 I think, so some tools have not caught up I guess). The exception is that irb can't support . on previous line because it attempts to eval it every time you press enter (for that reason we prefer . on previous line in our styleguide).

If that's not it, I personally am also confused by the ternary operator over multiple lines so I might reformat it like:

````
  def HintViewRequest.milestone_response(script, level, user)
    return [] unless HintViewRequest::ENABLED
    HintViewRequest.where(script: script, level: level, user: user)
      .pluck(:feedback_type, :feedback_xml)
      .map do |feedback_type, feedback_xml|
        {feedback_type: feedback_type, feedback_xml: feedback_xml}
    end
  end
end
````